tests/repository/test_yum_repository.py

```diff
@@ -193,3 +213,72 @@ def test_related_repositories_detached_client():
 
     with pytest.raises(DetachedException):
         repo_binary_test.get_binary_repository()
+
+
+def test_create_repository(client, requests_mocker):
+    repo = YumRepository(id="yum_repo")
+    repo.__dict__["_client"] = client
```

Comment:
I don't think this line should be here.  It's a hack done from within some tests to forcibly connect a repo object with a client, but the real users of the library aren't supposed to do that, and in this case it's important to test that `client.create_repository(repo)` actually works without doing this hack first.

I believe if you delete this line the test should still pass and it'd be more accurately covering what the user code should look like.